server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1268,7 +1265,27 @@ public static List<IndexTemplateMetadata> findV1Templates(Metadata metadata, Str
      */
     @Nullable
     public static String findV2Template(Metadata metadata, String indexName, boolean isHidden) {
-        final List<Tuple<String, ComposableIndexTemplate>> candidates = findV2CandidateTemplates(metadata, indexName, isHidden);
+        return findV2Template(metadata, metadata.templatesV2().entrySet(), indexName, isHidden, false);
+    }
```

Comment:
What if we create one more method called: `findV2TemplateFromSortedList` which will call `findV2Template(metadata, metadata.templatesV2().entrySet(), indexName, isHidden, true)` and we can document that if you want to retrieve the templates of multiple targets you can provide a sorted list of templates which will speed things up.

I think this will clarify a bit more which method to use and it will make explicit that the templates need to be ordered based on priority.